docs/upgrade.html

```diff
@@ -19,6 +19,10 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h4><a id="client_upgrade_4" href="#client_upgrade_4">Kafka Client upgrade path for 4.x</a></h4>
```

Comment:
This new section feels a bit weird because all the other section are per version. @ijuma and I discussed offline and we suggest to follow the same pattern for clients. Basically, we could have the following:

```
<h4><a id="upgrade_4_1_0" href="#upgrade_4_1_0">Upgrading Servers to 4.1.0 from any version 3.3.x through 4.0.x</a></h4>

<h4><a id="upgrade_4_1_0" href="#upgrade_clients_4_1_0">Upgrading Clients to 4.1.0</a></h4>

<h4><a id="upgrade_4_0_0" href="#upgrade_4_0_0">Upgrading Servers to 4.0.0 from any version 3.3.x through 3.9.x</a></h4>

Here we keep the current content.

<h4><a id="upgrade_4_0_0" href="#upgrade_clients_4_0_0">Upgrading Clients to 4.0.0 </a></h4>

Here we adds bit about rolling upgrade clients, similarly to what we have for the servers. Moreover, we add a not about the upgrade path and link the KIP.
```

We could follow this pattern for next releases. In the future, we could also consider moving the notable changes related to the clients to the clients section but this is not necessary for 4.0.

Reply:
@dajac Sounds great!

Do I need to make changes to the current title, and pave the way for the future?

```suggestion
<h4><a id="upgrade_clients_4_0_0" href="#upgrade_clients_4_0_0">Upgrading Clients to 4.0.</a></h4>
```

Or, it's all good for now.

Reply:
Yes, let's change the current title.

Reply:
@dajac 
![image](https://github.com/user-attachments/assets/3e573140-cdd8-4c9c-b61d-ccfcb5642308)

Reply:
@mingdaoy Thanks. I was perhaps not clear but I was trying to suggest a structure as follow:

```
<h4><a id="upgrade_4_0_0" href="#upgrade_4_0_0">Upgrading to 4.0.0</a></h4>

<h5><a id="upgrade_clients_4_0_0" href="#upgrade_clients_4_0_0">Upgrading Clients to 4.0.0</a></h5>

<p><b>For a rolling upgrade:</b></p>

<ol>
    <li>Upgrade the clients one at a time: shut down the client, update the code, and restart it.</li>
    <li>For the Kafka client upgrade path, note that many deprecated APIs were removed in Kafka 4.0. Additionally, upgrading directly to 4.x from certain versions is not feasible.
        For more information, please refer to <a href="https://cwiki.apache.org/confluence/x/y4kgF">KIP-1124</a>.</li>
</ol>

<h6><a id="upgrade_clients_400_notable" href="#upgrade_clients_400_notable">Notable changes in 4.0.0</a></h6>

<ul>
    <li>Please see notable changes in the server section.</li>
</ul>

<h5><a id="upgrade_servers_4_0_0" href="#upgrade_servers_4_0_0">Upgrading Servers to 4.0.0 from any version 3.3.x through 3.9.x</a></h5>

<p>Note: Apache Kafka 4.0 only supports KRaft mode - ZooKeeper mode has been removed. As such, <b>broker upgrades to 4.0.0 (and higher) require KRaft mode and
    the software and metadata versions must be at least 3.3.x</b> (the first version when KRaft mode was deemed production ready). For clusters in KRaft mode
    with versions older than 3.3.x, we recommend upgrading to 3.9.x before upgrading to 4.0.x. Clusters in ZooKeeper mode
    have to be <a href="/{{version}}/documentation.html#kraft_zk_migration">migrated to KRaft mode</a> before they can be upgraded to 4.0.x.
</p>

<p><b>For a rolling upgrade:</b></p>
....
```

We can do the same for the 4.1 section.

Reply:
![image](https://github.com/user-attachments/assets/b2178ff4-39ff-4988-abb4-0d4529345739)

Reply:
@dajac Please see above. Sorry for the wait.

Reply:
Thanks. Have you pushed the changes?

Reply:
@dajac pushed just now and synced with trunk